staging/src/k8s.io/cloud-provider/controllers/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -207,3 +207,26 @@ func shutdownInCloudProvider(ctx context.Context, cloud cloudprovider.Interface,
 
 	return shutdown, err
 }
+
+// ensureNodeExistsByProviderID checks if the instance exists by the provider id,
+// If provider id in spec is empty it calls instanceId with node name to get provider id
+func ensureNodeExistsByProviderID(ctx context.Context, instances cloudprovider.Instances, node *v1.Node) (bool, error) {
```

Comment:
I am not sure why we cannot use the same function in node_controller.go and when/where this func will be called?
If this func has to be added here, should the test also be added in node_lifecycle_controller_test.go?

Reply:
+1, seems like this would be a good start for a controller utils package under cloud-provider.

Reply:
A little duplication is acceptable here I think. I can't reuse ensureNodeExistsByProviderID from node_controller.go because it is in a separate package and ensureNodeExistsByProviderID is not widely used enough to justify a separate utils package for it either.